app/src/main/java/com/plaid/linksample/MainActivity.kt

```diff
@@ -61,41 +60,49 @@ class MainActivity : AppCompatActivity() {
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     setContentView(R.layout.activity_main)
-    setSupportActionBar(toolbar)
-    contentTextView = findViewById(R.id.content)
+    result = findViewById(R.id.result)
 
-    val fab = findViewById<FloatingActionButton>(R.id.open_link_fab)
+    val fab = findViewById<FloatingActionButton>(R.id.open_link)
     fab.setOnClickListener {
-      Plaid.setLinkEventListener(linkEventListener = {
-        Log.i("Event", it.toString())
-      })
-      Plaid.openLink(
-        activity = this,
-        linkConfiguration = LinkConfiguration(
-          clientName = "Test App",
-          products = listOf(PlaidProduct.TRANSACTIONS)
-        ),
-        requestCode = LINK_REQUEST_CODE
-      )
+      setOptionalEventListener()
+      openLink()
     }
   }
 
+  /**
+   * Optional, set an [event listener](https://plaid.com/docs/link/android/#handling-onevent).
+   */
+  private fun setOptionalEventListener() = Plaid.setLinkEventListener { event ->
+    Log.i("Event", event.toString())
+  }
+
+  private fun openLink() {
```

Comment:
Might be good to add a link to the parameters here plaid.com/docs/link/android/#parameter-reference